### ISSUE TITLE
There is a training loop in the trainer which return loss and accuracy for the last batch

### ISSUE THREAD

**Issue by burlachenkok** [Vision-CAIR/cs326-few-shot-classification#1]
https://github.com/Vision-CAIR/cs326-few-shot-classification/blob/7db7842ed30c14583bc7ab27ea4c2583620ae8b7/trainers/trainer.py#L46

I understand that we for save of computing power measure accuracy and loss via moving model, but it may be the case that current implementation that just returns last loss and last acc in the last batch a bit incorrect.

I suggest replace it: 
``` sum(losses)/len(losses), sum(accs)/len(accs)``

It's still approximate computation of loss and accuracy because we evaluate acc and loss in different points.

**Comment by universome** [comment]
To be honest, this design choice feels subjective. In the current implementation, we return a noisy estimate of the **final** training loss / **final** training accuracy of the episode. In your case, you propose to return the average training loss / average training accuracy. Since at the beginning of each episode the model has random performance, such measure would be polluted with bad scores received at the beginning of the training — especially if `num_train_steps_per_episode` is small (as in our case). For me, evaluating a model's performance based on its final performance feels more natural, but you can use whatever measure you are comfortable with
